xla/pjrt/pjrt_stream_executor_client.cc

```diff
@@ -2240,9 +2241,20 @@ std::unique_ptr<PjRtBuffer> OutputBufferHelper(
   std::shared_ptr<TrackedDeviceBuffer> out_buffer =
       TrackedDeviceBuffer::FromScopedShapedBuffer(result_buffer,
                                                   {definition_event});
+  Shape shape = result_buffer->on_device_shape();
+  PjRtMemorySpace* memory_space =
+      device->default_memory_space().value_or(nullptr);
```

Comment:
https://cs.opensource.google/tensorflow/tensorflow/+/master:third_party/xla/xla/pjrt/gpu/se_gpu_pjrt_client.cc;drc=9e0957281ee29229a2f907cad196a041c1b126d8;l=1144

So junwhan@ made it always returning a value. No need to .value_or(nullptr)?

Reply:
As far as I can see, the interpreter device can still return error, see https://github.com/openxla/xla/blob/60c9ecf92554fe5676b0358fecf93d13863f0476/xla/pjrt/interpreter_device.h#L26, https://github.com/openxla/xla/blob/60c9ecf92554fe5676b0358fecf93d13863f0476/xla/pjrt/pjrt_stream_executor_client.cc#L1364

No?